Fixed report command bug and changes default behaviour

Korne127 opened the pull request:
This pull request consists of two commits. I would recommend looking at those separately.
The first commit fixes a bug related to a setting in gtm. The second commit changes the default behaviour in that setting, so it reverts some of the changes in the first commit.
I think the change makes much sense (and I'll explain my reasoning here) but I did it that way so if you disagree, you could still use the first commit which just fixed the bug.

# Here is a short reasoning of the two commits:

## Fixed that -n=0 previously didn't work 
Previously, the limiter has been automatically set to zero. And in a case where no additional limiting flags have been present, it has been set to one.
That has created the bug that if someone has manually set the limiter to zero, that didn’t work, as there was no difference in setting it to zero and not setting it at all.

So I have changed the standard setting to -1. So now only if it has not explicitly been set to a different value (and remained at -1), it will get set to either zero or one (depending if there’s an additional limiter or not).
So now, if someone explicitly sets it to zero, that won’t change and it will work.

Additionally, if the project summary type is selected, all commits should be shown as well. Previously, there has been selected a high arbitrary number. While it works, it has described as „hack“ itself and is not very nice. Now I could also replace that just to setting it to zero which symbolises no limit (and won’t be changed in NewCommandLimiter anymore).

## Changed default setting to -n=0 
I've changed the default setting of n to zero. So now, unless explicitly stated differently, information for all commits will appear. I went through every single report type and think that this overall makes much more sense:

- Summary: With summary, it makes absolutely no sense to have n=1 as default. The summary just shows the time for each commit and with n=1 it just shows the number of the last commit, opposed to a real report of the time of each commit
- Project: The one type where it already has n=0 as default value
- Commits: The default value, also here I think it makes more sense to use n=0. The gtm report command reminds much of the git log command, which also doesn't just show information for the latest commit but for all existing. In general, if you want a report, that is supposed to be reporting the time spent on all commits. If someone just needs information for the latest one, n=0 shows it for every separated, so including the latest one. I would argue that having a report for all commits definitely is the expected and (in most cases) wanted behaviour.
- Files: n=0 makes more sense here as well. If you want to see on which files you spent how much time, in most cases you would want that for all commits obviously to see the time for each file. If someone just wants it for the latest commit, there is no real difference to the commit setting anyways.
- Timeline-Hours: I would argue in most cases n=0 is the expected behaviour to see when you've programmed how much and have that as a diagram. Just having it for the latest commit can be also interesting in rare occasions, but it doesn't make much sense to me as default behaviour.
- Timeline-Commits: Just using n=1 as default behaviour doesn't make much sense. That doesn't show a diagram but just one yellow bar - for the latest commit. Expected is a diagram of when you've made how many commits -> so n=0

In general, for the report command having a report of all commits and everything you've programmed in that project is the behaviour I think you'd want for most cases - it's also the standard for other programs like git itself with the logging command, so it's the expected behaviour as well. I just explicitly wrote it for the different categories to show that it makes sense in each one.

If you really don't agree with this (then I would find the reasoning for that interesting), you can of course only use the first commit which fixes the bug that you previously couldn't use n=0, but I think it would definitely improve the report command and it's simplicity to use both :)

Korne127 committed:
Fixed that -n=0 previously didn't work

Previously, n being explicitely set to zero (which removes a standard limiter in some cases) didn't work. The limiter has still been set to 1 if there was no additional date, author or message limiter.
This has been fixed now that the limiter works as documented.

Korne127 committed:
Changed default setting to -n=0

The default setting has been changed to 0 for the parameter n. That means that unless explicitely stated differently, the report information for all commits will be shown.

kilpkonn commented on command/report.go:
```diff
@@ -52,7 +52,7 @@ Options:
   Commit Limiting:
 
-  -n int=1                   Limit output, 0 is no limits, defaults to 1 when no limiting flags otherwise defaults to 0
+  -n int=1                   Limit output, 0 is no limit, defaults to 0
```
Change `-n int=1` -> `-n int=0`, as it is new default

Korne127 replied:
Oh, sorry, I completely oversaw that haha.

Korne127 committed:
Updated standard documentation value n=0

The documented standard value for n was previously still at one. It has been changed to zero now and the documentation of this setting has been adjusted to fit the general documentation style.
